packages/request-manager/src/controller.ts

```diff
@@ -0,0 +1,57 @@
+import { EventEmitter } from 'events';
+
+import { createTimeoutPromise } from '@trezor/utils';
+
+import { TorControlPort } from './torControlPort';
+
+interface TorConnectionOptions {
+    host: string;
+    port: number;
+    controlPort: number;
+    authFilePath: string;
+}
+
+export class TorController extends EventEmitter {
+    options: TorConnectionOptions;
+    controlPort: TorControlPort;
+    waitingTime = 1000;
+    maxTriesWaiting = 60;
+
+    constructor(options: TorConnectionOptions) {
+        super();
+        this.options = options;
+        this.controlPort = new TorControlPort({
+            host: this.options.host,
+            port: this.options.controlPort,
+            authFilePath: this.options.authFilePath,
+        });
+    }
+
+    async waitUntilAlive() {
+        let triesCount = 0;
+        const waitUntilResponse = async () => {
+            triesCount += 1;
+            if (triesCount === this.maxTriesWaiting) {
+                throw new Error('Timeout waiting for TOR control port');
+            }
+            try {
+                await this.controlPort.connect();
+                const isAlive = await this.controlPort.ping();
+                if (isAlive === true) {
+                    // It is running so let's not wait more.
+                    return;
+                }
+                await createTimeoutPromise(this.waitingTime);
+                await waitUntilResponse();
+            } catch (error) {
+                await createTimeoutPromise(this.waitingTime);
+                await waitUntilResponse();
```

Comment:
Wouldn't it be possible to unify these false/catch cases? Something like:

```
const isAlive = await this.controlPort.connect()
    .then(() => this.controlPort.ping())
    .catch(() => false)
if (!isAlive) {
    await createTimeoutPromise(this.waitingTime);
    await waitUntilResponse(triesCount - 1);
}
```

Reply:
I have been looking at it for a while and I think yes we save couple of lines of code but I think we lose readability.

I would prefer readability. But of course this is open to different opinions :)

Reply:
@marekrjpolak after our conversation where you found out that the previous implementation could lead to some issues I am following your suggestion as you can see 6e9f16a16d865b7669e443e3792c7012c182e103 with some little changes.

The more I look at it the more readable I find it :sweat_smile: